Parkinson/TasksAndSteps/APHParkinsonActivityViewController.m

```diff
@@ -0,0 +1,254 @@
+                                                           answerFormat:format];
+            [stepQuestions addObject:item];
+        }
+        [step setFormItems:stepQuestions];
+        
+        NSMutableArray  *newSteps = [task.steps mutableCopy];
+        if ([newSteps count] >= 1) {
+            [newSteps insertObject:step atIndex:1];
+        }
+        replacementTask = [[ORKOrderedTask alloc] initWithIdentifier:taskTitle steps:newSteps];
+    }
+    return  replacementTask;
+}
+
+#pragma  mark  -  Step View Controller Delegate Methods
+
+- (void)stepViewControllerResultDidChange:(ORKStepViewController *)stepViewController
+{
+    NSString  *stepIdentifier = stepViewController.step.identifier;
+    if ([stepIdentifier isEqualToString:kMomentInDayStepIdentifier] == YES) {
+        ORKStepResult  *stepResult = stepViewController.result;
+        if ([stepResult.results count] > 0) {
+            id  object = [stepResult.results lastObject];
+            if ([object isKindOfClass:[ORKChoiceQuestionResult class]] == YES) {
+                ORKChoiceQuestionResult  *result = (ORKChoiceQuestionResult *)object;
+                    //
+                    //  stash the object away
+                    //
+                if (result.choiceAnswers.lastObject != nil) {
+                    NSDictionary  *dictionary = @{
+                                                  kMomentInDayChoiceAnswerKey : result.choiceAnswers.lastObject,
+                                                  kMomentInDayQuestionTypeKey : @(result.questionType),
+                                                  kMomentInDayIdentifierKey   : result.identifier,
+                                                  kMomentInDayStartDateKey    : result.startDate,
+                                                  kMomentInDayEndDateKey      : result.endDate,
+                                                  };
+                    NSUserDefaults  *defaults = [NSUserDefaults standardUserDefaults];
+                    [defaults setObject:dictionary forKey:kMomentInDayUserDefaultsKey];
+                    [defaults synchronize];
+                }
+            }
+        }
+    }
+    if ([super respondsToSelector:@selector(stepViewControllerResultDidChange:)] == YES) {
+        [super stepViewControllerResultDidChange:stepViewController];
+    }
+}
+
+#pragma  mark  -  Step View Controller Delegate Methods
+
+- (void)taskViewController:(ORKTaskViewController *)taskViewController didFinishWithReason:(ORKTaskViewControllerFinishReason)reason error:(nullable NSError *)error
+{
+    if (reason == ORKTaskViewControllerFinishReasonCompleted) {
+        BOOL  foundSurveyStep = NO;
+        
+        for (ORKStepResult *stepResult in self.result.results) {
+            if ([stepResult.identifier isEqualToString:kMomentInDayStepIdentifier] == YES) {
+                foundSurveyStep = YES;
+                break;
+            }
+        }
```

Comment:
> On Jun 29, 2015, at 2:45 PM, Yuan Zhu <notifications@github.com> wrote:
> 
> In Parkinson/TasksAndSteps/APHParkinsonActivityViewController.m <https://github.com/ResearchKit/mPower/pull/6#discussion_r33517507>:
> 
> > +    }
> > +}
> > +
> > +#pragma  mark  -  Step View Controller Delegate Methods
> > +
> > +- (void)taskViewController:(ORKTaskViewController *)taskViewController didFinishWithReason:(ORKTaskViewControllerFinishReason)reason error:(nullable NSError *)error
> > +{
> > +    if (reason == ORKTaskViewControllerFinishReasonCompleted) {
> > +        BOOL  foundSurveyStep = NO;
> > +        
> > +        for (ORKStepResult *stepResult in self.result.results) {
> > +            if ([stepResult.identifier isEqualToString:kMomentInDayStepIdentifier] == YES) {
> > +                foundSurveyStep = YES;
> > +                break;
> > +            }
> > +        }
> ORKTaskResult is a subclass ORKCollectionResult
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ResearchKit/mPower/pull/6/files#r33517507>.
> 


Okay, did that.  Thank you for the reminder—I’d forgotten all about that method.